cli/lib/nf3.mjs

```diff
@@ -1298,7 +1298,19 @@ class Nf3 {
             );
             challengeEmitter.emit('receipt', receipt, type, txSelector);
           } catch (err) {
-            challengeEmitter.emit('error', err, type, txSelector);
+            logger.error({
+              msg: 'Error while trying to challenge a block',
+              type,
+              err,
+            });
+            try {
+              challengeEmitter.emit('error', err, type, txSelector);
+            } catch (err2) {
```

Comment:
I'm not sure if this will work, when one emits an `error`, the Emitter manager will try to verify if there is any listener for `error`, if not it will log the error and will stop the process - it doesn't throw an error. The best way would be to use the method `createEmitter` when creating the `challengerEmitter` (similar to the proposer), then the server would not stop if the emitter emits an `error`.